enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/interceptor/MediationPolicyFilter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.choreo.connect.enforcer.interceptor;
+
+import io.grpc.netty.shaded.io.netty.handler.codec.http.HttpMethod;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.choreo.connect.enforcer.commons.Filter;
+import org.wso2.choreo.connect.enforcer.commons.model.Policy;
+import org.wso2.choreo.connect.enforcer.commons.model.PolicyConfig;
+import org.wso2.choreo.connect.enforcer.commons.model.RequestContext;
+
+import java.util.Map;
+
+/**
+ * Apply mediation policies.
+ */
+public class MediationPolicyFilter implements Filter {
+    private static final Logger log = LogManager.getLogger(MediationPolicyFilter.class);
+    @Override
+    public boolean handleRequest(RequestContext requestContext) {
+        // get operation policies
+        PolicyConfig policyConfig = requestContext.getMatchedResourcePath().getPolicyConfig();
+        // apply in policies
+        if (policyConfig.getIn() != null && policyConfig.getIn().size() > 0) {
+            for (Policy policy : policyConfig.getIn()) {
+                applyPolicy(requestContext, policy);
+            }
+        }
+        return true;
+    }
+
+    private void applyPolicy(RequestContext requestContext, Policy policy) {
+        //todo(amali) check policy order
+        switch (policy.getTemplateName()) {
+            case "SET_HEADER":
+            case "RENAME_HEADER": {
```

Comment:
In the design doc, there isn't a "RENAME_HEADER" option. Do we truly need it. As in, renaming a header would mean something like the following right?

- old header -> "aaa: header_value"
- new header -> "bbb: header_value"

Reply:
I rechecked. There is a policy called rename header.